src/scripts/main.js

```diff
@@ -1 +1,80 @@
 'use strict';
+
+document.addEventListener('contextmenu', (e) => e.preventDefault());
+
+const showNotification = (message, type) => {
+  document.body.insertAdjacentHTML(
+    'afterbegin',
+    `<div class="${type}" data-qa="notification">${message}</div>`,
+  );
+};
+
+const success = (message) => showNotification(message, 'success');
+const error = (message) => showNotification(message, 'error');
+
+const firstPromise = new Promise((resolve, reject) => {
+  const clickHandler = (e) => {
+    if (e.button === 0) {
+      resolve();
+      document.removeEventListener('click', clickHandler);
+    }
+  };
+
+  document.addEventListener('click', clickHandler);
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+    document.removeEventListener('click', clickHandler);
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  const clickHandler = (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve();
+      document.removeEventListener('click', clickHandler);
+      document.removeEventListener('contextmenu', clickHandler);
```

Comment:
The 'contextmenu' event listener is removed here, but it was added in line 41. Ensure that the logic for removing event listeners is consistent with where they are added.